src/manifest.c

```diff
@@ -481,6 +469,7 @@ static int try_delta_manifest_download(int current, int new, char *component, st
 	}
 
 	unlink(deltafile);
+	populate_file_struct(file, newfile);
 	ret = compute_hash(file, newfile); // MUST save new hash!
```

Comment:
Could we pass in the file struct as a pointer to a pointer so that if this function fails it won't have modified the file struct passed in and if it succeeds can modify the owning pointer to point to the new file struct?